client/src/pages/Transactions/Transactions.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { useState, useEffect, createContext } from "react";
+import { Container } from "./Transactions.styles";
+import { TransactionItemAdder, TransactionItemsList } from "../../components";
+import { AnimatePresence } from "framer-motion";
+import { getAllItems } from "../../API/TransactionMethods";
+import { ItemData } from "../../constants";
+
+const PathContext = createContext<string>("");
+
+interface TransactionProps {
+  pageType: "income" | "expense";
+}
+
+const Income: React.FC<TransactionProps> = ({ pageType }) => {
+  const [displayAdder, setDisplayAdder] = useState<boolean>(false);
+  const [rerender, setRerender] = useState<boolean>(false);
+  const [filteredIncomeItems, setFilteredIncomeItems] = useState<
```

Comment:
Couldn't this just be filteredTransactionItems?

Maybe an object with 2 arrays of items in it?

Const [filteredTransactionItems, setFilteredTransactionItems] = useState({expenses: [], incomes: []});